service/controller/resource/cleanuprecordsets/delete.go

```diff
@@ -0,0 +1,115 @@
+package cleanuprecordsets
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/controller/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var hostedZones []*route53.HostedZone
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding hosted zones for tenant cluster %#q", key.ClusterID(&cr)))
+
+		hostedZonesInput := &route53.ListHostedZonesByNameInput{}
+
+		o, err := cc.Client.TenantCluster.AWS.Route53.ListHostedZonesByName(hostedZonesInput)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		for _, zone := range o.HostedZones {
+			baseDomain := fmt.Sprintf("%s.", key.TenantClusterBaseDomain(cr))
```

Comment:
This doesn't look right to me. For new style clusters we don't have `k8s.*` zone. Each cluster has separate `CLUSTER_ID.k8s.*` zone in the BYOC account. This zone should be deleted by the CF stack. If dynamic DNS is configured correctly (i.e. creating records in the new zone) then this resources should not be needed.

Reply:
I didn’t get this

CF can’t delete hosted zone if there are records not from CF. Who does cleanup then?

Reply:
> CF can’t delete hosted zone if there are records not from CF.

I didn't know this. Then we need the resource.

But still we shouldn't create any records in `k8s` zone. We should create them in `CLUSTER_ID.k8s` zone.

Reply:
I’m again not sure about what you’re talking here. Resources created in clusterID.k8s zones (outside the stack) 
This resource cleane those resources in clusterID.k8s zone